src/notifications/NotificationsHandler.tsx

```diff
@@ -372,12 +345,35 @@ export const NotificationsHandler = ({ walletReady }: Props) => {
     }
   }, [handleDeferredNotificationIfNeeded, prevWalletReady, walletReady]);
 
+  /*
+   * Initializing MMKV storage with empty settings for all wallets that weren't yet initialized
+   * and start subscribing for notifications with Firebase
+   */
   useEffect(() => {
-    addresses.forEach(({ address, relationship }) => {
-      dispatch(notificationsSubscription(address));
-      addDefaultNotificationSettingsForWallet(address, relationship);
-    });
-  }, [addresses, dispatch]);
+    // It is run only once per app session, that means on every app cold start
```

Comment:
This sounds like it should be a migration? You added it to the wallet creation flows, so if we migrate old accounts and handle new accounts there, do we need this?

Reply:
@estrattonbailey I think this is the only outstanding thing. If you feel there's a way to move it under migrations with the constraints above I'm happy to chat about it.